Error of input text not matching decoded output of tokenizer

## 🐛 Bug Report

If text is used as input to the `attribute` method that contains spaces before non-alphanumeric characters, the decoding of the tokenizer does not match the input anymore, leading to the assert error.

## 🔬 How To Reproduce

Give as input a text containing a special character (e.g.: `.` or `?`) preceded by a white space while using a GPT-like model (and tokenizer).

### Code sample
Steps to reproduce the behavior:

```python
import inseq
model = inseq.load_model('gpt2', attribution_method='input_x_gradient') # Or any other gpt-like model
model.attribute(input_texts='Hello . This is an example')
```
Returns the following error: ```AssertionError: Forced generations with decoder-only models must start with the input texts.```


### Environment

* OS: macOS (but independent from the OS)
* Python version: `Python 3.9.15`
* Inseq version, get it with: `0.4.0`

## Expected behavior

The decoded text output from the tokenizer should be identical to the input text, allowing the assert to be correctly verified:
```python
assert all(
    generated_texts[idx].startswith(input_texts[idx]) for idx in range(len(input_texts))
), "Forced generations with decoder-only models must start with the input texts."
```

## Additional context

The issue is related to the type of tokenizer used, already reported in huggingface/transformers#21119. To solve the problem, it is recommended to use the `clean_up_tokenization_spaces=False` flag when decoding the text.

--- a/inseq/models/huggingface_model.py
+++ b/inseq/models/huggingface_model.py
@@ -215,10 +215,7 @@ class HuggingfaceModel(AttributionModel):
             **kwargs,
         )
         sequences = generation_out.sequences
-        texts = self.tokenizer.batch_decode(
-            sequences,
-            skip_special_tokens=True,
-        )
+        texts = self.decode(ids=sequences, skip_special_tokens=True)
         if return_generation_output:
             return texts, generation_out
         return texts
@@ -295,7 +292,11 @@ class HuggingfaceModel(AttributionModel):
         ids: Union[List[int], List[List[int]], IdsTensor],
         skip_special_tokens: bool = True,
     ) -> List[str]:
-        return self.tokenizer.batch_decode(ids, skip_special_tokens=skip_special_tokens)
+        return self.tokenizer.batch_decode(
+            ids,
+            skip_special_tokens=skip_special_tokens,
+            clean_up_tokenization_spaces=False,
+        )
 
     def embed_ids(self, ids: IdsTensor, as_targets: bool = False) -> EmbeddingsTensor:
         if as_targets and not self.is_encoder_decoder:

